plat/socionext/synquacer/sq_bl31_setup.c

```diff
@@ -44,13 +44,18 @@ uint32_t sq_get_spsr_for_bl33_entry(void)
 	unsigned long el_status;
 	unsigned int mode;
 	uint32_t spsr;
+	uint32_t *gpio = (uint32_t *)PLAT_SQ_GPIO_BASE;
 
 	/* Figure out what mode we enter the non-secure world in */
 	el_status = read_id_aa64pfr0_el1() >> ID_AA64PFR0_EL2_SHIFT;
 	el_status &= ID_AA64PFR0_ELX_MASK;
 
 	mode = (el_status) ? MODE_EL2 : MODE_EL1;
 
+	/* enable the stage 2 override page tables if PD[1] is set */
+	if (gpio[0] & 0x2)
+		mode = MODE_EL1;
+
```

Comment:
Ok, so you can select EL1 or EL2 based on the signal from a GPIO pin. However, you always set up the stage 2 tables. If BL33 gets control at EL2, it can replace them, making the call to `plat_setup_s2_translation()` useless. Wouldn't it be better to map them only if you are entering BL33 in EL1?

Reply:
Thanks for your comment. Actually this is a workaround for PCIe for current version of SC2A11 SoC. After more discussion on this internally, it seems that this bug in hardware will be fixed in upcoming SoC. So internally we reached to a conclusion to not upstream this workaround due to its future maintainability. I will remove these workaround patches from this PR.

Reply:
Ok. Just out of curiosity, did you create a tool to generate the translation tables?

Reply:
@ardbiesheuvel Can you please help answer above query?

Reply:
I was very clear when we discussed this offline: this functionality does *not* belong in the upstream port. It works around a PCIe issue in rev 1 silicon, and it has its own share of issues.

Reply:
Ah, that wasn't the question - apologies.

The stage 2 tables are created statically in NOR flash

https://git.linaro.org/leg/noupstream/edk2-platforms.git/tree/Silicon/Socionext/SynQuacer/Stage2Tables/Stage2Tables.S?h=developer-box&id=ba32985a0631ef371b06cdbfe53f4dd6d11557ba

Reply:
Thanks @ardbiesheuvel

Reply:
Oh, ok, so there's no tool. You simply generate the tables by hand. Still interesting, thanks.